kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/EnvelopeEncryption.java

```diff
@@ -61,33 +62,41 @@ record KmsCacheConfig(
 
     }
 
+    public record SharedServices<K, E>(
```

Comment:
The name implies something ultra generic by the generic types imply something quite encryption specific, well key value specific really. 

Longer term I think the kroxylicious api/runtime should be providing some thing to share state between filter instances but I'm keen to keep it local to topic encryption for now. 

So I think I err towards a more specific name in this case, but probably using `SharedService` or something for the runtime provided version.

Reply:
`SharedEncryptionState` with a Javadoc making it clear that these things are shared between each filter instance and so need to be safe for sharing between threads?

Reply:
Yeah, our convention has been a `context` class rather than a `state` class as I think they represent the same things but happy to hear opposing views.

Reply:
I think I'm going with `SharedEncryptionContext`, fwiw.